src/server/cmd/job-shim/main.go

```diff
@@ -30,6 +38,74 @@ func main() {
 	env.Main(do, &appEnv{})
 }
 
+func downloadInput(c *client.APIClient, commitMounts []*fuse.CommitMount) error {
+	for _, commitMount := range commitMounts {
+		if commitMount.Alias == "prev" || commitMount.Alias == "out" {
+			continue
+		}
+		repo := commitMount.Commit.Repo.Name
+		commitID := commitMount.Commit.ID
+		var fromCommitID string
+		if commitMount.DiffMethod != nil && commitMount.DiffMethod.FromCommit != nil {
+			fromCommitID = commitMount.DiffMethod.FromCommit.ID
+		}
+		var fullFile bool
+		if commitMount.DiffMethod != nil {
+			fullFile = commitMount.DiffMethod.FullFile
+		}
+		shard := commitMount.Shard
+		fileInfos, err := c.ListFile(repo, commitID, "/", fromCommitID, fullFile, shard, false)
+		if err != nil {
+			return err
+		}
+		for _, fileInfo := range fileInfos {
+			if err := os.MkdirAll(filepath.Join(PFSInputPrefix, repo), 0777); err != nil {
+				return err
+			}
+			path := filepath.Join(PFSInputPrefix, repo, fileInfo.File.Path)
+			f, err := os.Create(path)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
```

Comment:
You're swallowing errors from `f.Close()` here. You can report them with a closure. Look at this for an [example](https://github.com/pachyderm/pachyderm/blob/c9c44b8429a2676272c632c94cb43e0e642ca0d1/src/server/pfs/server/obj_block_api_server.go#L62).